packages/slider/src/useSlider.ts

```diff
@@ -0,0 +1,292 @@
+    ]
+  );
+
+  const getThumbProps = useCallback(
+    (thumb: 'min' | 'max'): IUseSliderReturnValue['getMinThumbProps'] =>
+      ({ onKeyDown, onMouseDown, ...other }) => {
+        const handleKeyDown = (event: KeyboardEvent) => {
+          if (!disabled) {
+            let value;
+
+            switch (event.key) {
+              case KEYS.RIGHT:
+                value =
+                  (thumb === 'min' ? position.minValue : position.maxValue) + (rtl ? -step : step);
+                break;
+
+              case KEYS.UP:
+                value = (thumb === 'min' ? position.minValue : position.maxValue) + step;
+                break;
+
+              case KEYS.LEFT:
+                value =
+                  (thumb === 'min' ? position.minValue : position.maxValue) - (rtl ? -step : step);
+                break;
+
+              case KEYS.DOWN:
+                value = (thumb === 'min' ? position.minValue : position.maxValue) - step;
+                break;
+
+              case KEYS.PAGE_UP:
+                value = (thumb === 'min' ? position.minValue : position.maxValue) + jump;
+                break;
+
+              case KEYS.PAGE_DOWN:
+                value = (thumb === 'min' ? position.minValue : position.maxValue) - jump;
+                break;
+
+              case KEYS.HOME:
+                value = min;
+                break;
+
+              case KEYS.END:
+                value = max;
+                break;
+            }
+
+            if (value !== undefined) {
+              event.stopPropagation();
+              setThumbPosition(thumb)(value);
+            }
+          }
+        };
+
+        const handleMouseMove = (event: MouseEvent) => {
+          const value = getTrackPosition(event);
+
+          if (value !== undefined) {
+            setThumbPosition(thumb)(value);
+          }
+        };
+
+        const handleMouseUp = () => {
+          doc.removeEventListener('mousemove', handleMouseMove);
+          doc.removeEventListener('mouseup', handleMouseUp);
+        };
+
+        const handleMouseDown = (event: MouseEvent) => {
+          if (!disabled && event.button === 0) {
+            event.stopPropagation();
+            doc.addEventListener('mousemove', handleMouseMove);
+            doc.addEventListener('mouseup', handleMouseUp);
+            event.target && (event.target as HTMLElement).focus();
+          }
+        };
+
+        return {
+          'data-garden-container-id': 'containers.slider.thumb',
+          'data-garden-container-version': PACKAGE_VERSION,
+          role: 'slider',
+          tabIndex: disabled ? -1 : 0,
+          'aria-valuemin': thumb === 'min' ? min : position.minValue,
+          'aria-valuemax': thumb === 'min' ? position.maxValue : max,
+          'aria-valuenow': thumb === 'min' ? position.minValue : position.maxValue,
+          onKeyDown: composeEventHandlers(onKeyDown, handleKeyDown),
+          onMouseDown: composeEventHandlers(onMouseDown, handleMouseDown),
```

Comment:
Is it worth attaching these event listeners to touch events, too? (In case a user is interacting with the range on a touch screen.)

Reply:
I think that would be a worthwhile enhancement to the code ported over from `MultiThumbRange`. I'll take a closer look :eyes:.